lib/ReactViews/Tools/ToolModal.tsx

```diff
@@ -0,0 +1,133 @@
+import { TFunction } from "i18next";
+import { observer } from "mobx-react";
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import styled, { useTheme } from "styled-components";
+import ViewState from "../../ReactViewModels/ViewState";
+import { GLYPHS, StyledIcon } from "../Icon";
+
+const Box = require("../../Styled/Box").default;
+const RawButton: any = require("../../Styled/Button").RawButton;
+const Spacing: any = require("../../Styled/Spacing").default;
+const Text: any = require("../../Styled/Text").default;
+
+export interface FrameProps {
+  title: string;
+  viewState: ViewState;
+}
+
+export const Frame: React.FC<FrameProps> = observer(props => {
+  const theme = useTheme();
+  const [t] = useTranslation();
+  const [showChildren, setShowChildren] = useState(true);
+  const { viewState } = props;
+  return (
+    <Wrapper isMapFullScreen={viewState.isMapFullScreen}>
+      <Toggle
+        paddedVertically
+        paddedHorizontally={2}
+        centered
+        justifySpaceBetween
+        backgroundColor={theme.toolPrimaryColor}
+      >
+        <Title title={props.title} icon={GLYPHS.search} />
+        {/* margin-right 5px for the padded button offset - larger click area
+          but visible should be inline with rest of box */}
+        <Box centered css={"margin-right:-5px;"}>
+          <ToolCloseButton viewState={viewState} t={t} />
+          <Spacing right={4} />
+          {/* collapse button */}
+          <RawButton onClick={() => setShowChildren(!showChildren)}>
+            <Box paddedRatio={1} centered>
+              <StyledIcon
+                styledWidth="12px"
+                light
+                glyph={showChildren ? GLYPHS.opened : GLYPHS.closed}
+              />
+            </Box>
+          </RawButton>
+        </Box>
+      </Toggle>
+      {showChildren && props.children}
+    </Wrapper>
+  );
+});
+
+export const Main = styled(Text)`
+  display: flex;
+  flex-direction: column;
+  padding: 15px;
+  overflow-y: auto;
+  ${({ theme }) => theme.borderRadiusBottom(theme.radius40Button)}
+  background-color: ${p => p.theme.darkWithOverlay};
+  min-height: 350px;
+`;
+
+const Wrapper = styled(Box).attrs({
+  column: true,
+  positionAbsolute: true,
+  styledWidth: "340px"
+  // charcoalGreyBg: true
+})`
+  top: 70px;
+  left: 0px;
+  min-height: 220px;
+  // background: ${p => p.theme.dark};
+  margin-left: ${props =>
+    props.isMapFullScreen ? 16 : parseInt(props.theme.workbenchWidth) + 40}px;
+  // TODO: Transitioning on margin-left incurs an expensive layout re-calculation, consider disabling
+  transition: margin-left 0.25s;
```

Comment:
All TODOs need to be either resolved or have an Issue created fo them in GitHub so that they don't get lost. We kinda let this go a bit while we were refactoring the model layer, but we're nearly done :)

Reply:
This comes from code I extracted from DiffTool. I have removed the TODO anyways as it doesn't seem like it is causing too much lag.